### ISSUE TITLE
Need for null target

### ISSUE THREAD

**Issue by stevefal** [jtackaberry/flexrouter#5]
Sometimes a keyswitch needs to block notes on a certain channel. If there is no patch for a particular articulation and a given part, then those note events need to go nowhere. I can send them to an unused channel like D16, but I'd rather have a "block" or null channel to send to.

For me, "0" would me more useful as a null channel than "use default target channel for rule" which I have so far never used. I assume that modality pertains to cases where CCs are converted to keyswitch notes.

I wonder if Target would ever equal "default target" in the "Block" case. For me the "Block" case is about rerouting events to different MIDI channels. What else would it be for?

**Comment by jtackaberry** [comment]
Yes, that's the primary (and maybe only) use-case for the block keyswitch action.  So your implication that target channel of 0 could mean a null (blackhole) target instead of the rule default target makes sense, except the next question is whether the use-case behind a null target could also apply to the passthrough or redirect cases.

With respect to the purpose behind a rule default target channel, your intuition is not far off: the real use for this is in the keyswitch remapping use-cases (not just CCs mind you, but any redirection).  For example, maybe I have one instrument on channel A1 and another on channel B1, and I just want to use FlexRouter for remapping the keyswitches, not for routing MIDI events.  In this case, I would have two rules (one for each instrument), and by having a default target channel, it allows me to move the instruments to different channels without having to reconfigure each keyswitch one by one.

That may be an esoteric use-case, but it's no less esoteric than those needing a null target. :)

I suppose the simple solution would be to allow say -1 and have that mean null target.

**Comment by jtackaberry** [comment]
Now that I think about it, I guess if you redirect or passthrough a keyswitch, it's because there's some patch there to receive it.  Why would you want to passthrough/redirect a keyswitch to a patch but not the subsequent MIDI events on the source channel?

So that said, maybe a context dependent redefinition for target 0 with the block action to mean null target.  It may not be terribly intuitive, but it should be easy to implement.

**Comment by stevefal** [comment]
-1 would work, I suppose, but I feel that both 0 and -1 having overloaded meanings begins a slippery slope toward arcane UI. I'm OK using D16 for now, pending more thought and feedback. For me "Block" by itself is already elusive language in that it pertains only to the keyswitch event and not the events being routed. So in that vain, I think the UI language is already a bit terse for the depth of concepts addressed.

I just read your intervening post. Again I'm able to hold tight on D16 to prevent too much churn in the short term. Perhaps other matters will affect the outcome of this question.

**Comment by jtackaberry** [comment]
Suggestions on how to make the UI more intuitive are quite welcome.

**Comment by stevefal** [comment]
I'll give it some thought.

**Comment by jtackaberry** [comment]
I decided to change all the channel fields to a menu list.  This has some disadvantages, but the obvious advantage is that I can include other options (like null, omni, default, etc.) without cluttering the UI.

I've implemented a null target which will be included in the upcoming 1.1 release.

Here's the UI as it currently is in my local tree.  Let me know if you have any suggestions.

![flexrouter11](https://cloud.githubusercontent.com/assets/1280551/14892631/b48429f6-0d39-11e6-8ee0-6a39f6fbba2d.png)
